src/main/java/com/unstoppabledomains/resolution/Utilities.java

```diff
@@ -1,8 +1,31 @@
 package com.unstoppabledomains.resolution;
 
+import com.unstoppabledomains.resolution.artifacts.Numeric;
+
+import org.bouncycastle.crypto.digests.SHA256Digest;
+
 public class Utilities {
+  private Utilities() {}
+
   public static boolean isNull(String value) {
     return ( value == null || value.equals("0x0000000000000000000000000000000000000000") || value.equals("0x")
         || value.equals("") || value.length() == 0);
   }
+
+  public static String sha256(String key) {
+    return sha256(key, false);
+  }
+
+  public static String sha256(String key, boolean hexEncoding) {
```

Comment:
I had to do this so I could hash it out in a single line

```java
Optional<String> namehash = Arrays.stream(labels)
.reduce((parent, label) -> Utilities.sha256(parent + Utilities.sha256(label), true)); 
```

The hex encoding is needed in the first sha256 (from left to right) and doesn't needed in the second

 I do have methods hextoString and stringToHex in Numeric class as I am using one inside the sha256

```java
if (hexEncoding) {
        digester.update(Numeric.hexStringToByteArray(key), 0, Numeric.hexStringToByteArray(key).length);
    } else {
        digester.update(key.getBytes(), 0, key.length());
    }
```

We also use the similar feature in our main resolution library: 

```javascript
sha256(parent + sha256(label, { hexPrefix: false }), {
      hexPrefix: options.prefix,
      inputEnc: 'hex',
    });
```

This is done to consider a hash as a hex number rather than an array of chars.